src/DGtal/base/IntegerSequenceIterator.h

```diff
@@ -0,0 +1,167 @@
+/**
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Lesser General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ **/
+
+#pragma once
+
+/**
+ * @file
+ * @author Jacques-Olivier Lachaud (\c jacques-olivier.lachaud@univ-savoie.fr )
+ * Laboratory of Mathematics (CNRS, UMR 5127), University of Savoie, France
+ *
+ * @date 2017/05/28
+ *
+ * Header file for module IntegerSequenceIterator.cpp
+ *
+ * This file is part of the DGtal library.
+ */
+
+#if defined(IntegerSequenceIterator_RECURSES)
+#error Recursive header files inclusion detected in IntegerSequenceIterator.h
+#else // defined(IntegerSequenceIterator_RECURSES)
+/** Prevents recursive inclusion of headers. */
+#define IntegerSequenceIterator_RECURSES
+
+#if !defined IntegerSequenceIterator_h
+/** Prevents repeated inclusion of headers. */
+#define IntegerSequenceIterator_h
+
+//////////////////////////////////////////////////////////////////////////////
+// Inclusions
+#include <iostream>
+#include <boost/iterator/iterator_facade.hpp>
+#include "DGtal/base/Common.h"
+//////////////////////////////////////////////////////////////////////////////
+
+namespace DGtal
+{
+
+  /////////////////////////////////////////////////////////////////////////////
+  // template class IntegerSequenceIterator
+  /**
+   * Description of template class 'IntegerSequenceIterator' <p>
+   * \brief Aim: It is a simple class that mimics a (non mutable)
+   * iterator over integers. You can increment it, decrement it,
+   * displace it, compare it, etc. It is useful if you have a
+   * collection of consecutive integers, and you wish to create an
+   * iterator over it. It is used in the class TriangulatedSurface for
+   * example, since vertices are numbers from 0 to nbVertices - 1.
+   */
```

Comment:
At this point, I don't see why you need this but it would probably make sense later..

Is this class model of some concepts we have on iterators ? If so, please update the doc.